### PR TITLE
Move a resize from codegen to code.

### PR DESCRIPTION
**Pull request by gchanan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#33024 Move a resize from codegen to code.**

Differential Revision: [D19774147](https://our.internmc.facebook.com/intern/diff/D19774147)

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Is the goal here to delete the `resize:` from the TH wrapping codegen? 

Also you seem to stop using the `_th_gather` function. Can that be deleted? Or the codegen generates both `_th_gather` and `_th_gather_out` from the same entry?

**Comment by gchanan** [comment]
yes, the goal is to remove "resize" from the wrapping code.

And yes, the codegen generates both _th_gather and _th_gather_out from the same entry.  I've debated adding an option to stop codegening one or the other, but not sure it's worth it.

File: aten/src/ATen/native/cuda/LegacyDefinitions.cpp [hunk 1]
```diff
@@ -80,11 +80,13 @@ Tensor & masked_select_out_cuda(Tensor & result, const Tensor & self, const Tens
 }
 
 Tensor & gather_out_cuda(Tensor & result, const Tensor & self, int64_t dim, const Tensor & index, bool sparse_grad) {
+  result.resize_(index.sizes());
```
Comment (albanD) [inline aten/src/ATen/native/cuda/LegacyDefinitions.cpp]: nit: is there any type-check necessary here (like the one we discussed for the cummax implementation)?

**Review by albanD — APPROVED** [review]
LGTM

File: aten/src/ATen/native/cuda/LegacyDefinitions.cpp (hunk 1 above)
Comment (albanD) [reply aten/src/ATen/native/cuda/LegacyDefinitions.cpp]: After offline chat. This is not needed as type checks are done in the THC binding!

**Pull request closed without merging**